src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts

```diff
@@ -83,16 +88,81 @@ export class DatasetService {
     }
   }
 
-  public fetchOptions(
+  public async fetchOptions(
     services: IDataPluginServices,
     path: DataStructure[],
     dataType: string
   ): Promise<DataStructure> {
     const type = this.typesRegistry.get(dataType);
     if (!type) {
-      throw new Error(`No handler found for type: ${path[0]}`);
+      throw new Error(`No handler found for type: ${dataType}`);
     }
-    return type.fetch(services, path);
+
+    const lastPathItem = path[path.length - 1];
+    const cacheKey = `${dataType}.${lastPathItem.id}`;
+
+    const cachedDataStructure = this.sessionStorage.get<CachedDataStructure>(cacheKey);
+    if (cachedDataStructure.children?.length > 0) {
```

Comment:
What if the datastructure has no children?

Reply:
should attempt the fetch correct? we could check `hasNext` but to me in makes sense to refetch always if no children.